contracts/tokenbridge/arbitrum/L2AtomicTokenBridgeFactory.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+import {L2GatewayRouter} from "./gateway/L2GatewayRouter.sol";
+import {L2ERC20Gateway} from "./gateway/L2ERC20Gateway.sol";
+import {L2CustomGateway} from "./gateway/L2CustomGateway.sol";
+import {L2WethGateway} from "./gateway/L2WethGateway.sol";
+import {StandardArbERC20} from "./StandardArbERC20.sol";
+import {BeaconProxyFactory} from "../libraries/ClonableBeaconProxy.sol";
+import {aeWETH} from "../libraries/aeWETH.sol";
+import {UpgradeableBeacon} from "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {
+    TransparentUpgradeableProxy,
+    ITransparentUpgradeableProxy
+} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+/**
+ * @title Layer2 token bridge creator
+ * @notice This contract is used to deploy token bridge on L2 chain.
+ * @dev L1AtomicTokenBridgeCreator shall call `deployL2Contracts` using retryable and that will result in deployment of canonical token bridge contracts.
+ */
+contract L2AtomicTokenBridgeFactory {
+    function deployL2Contracts(
+        L2RuntimeCode calldata l2Code,
+        address l1Router,
+        address l1StandardGateway,
+        address l1CustomGateway,
+        address l1WethGateway,
+        address l1Weth,
+        address l2StandardGatewayCanonicalAddress,
+        address rollupOwner
+    ) external {
+        // create proxyAdmin which will be used for all contracts
+        address proxyAdmin = address(new ProxyAdmin{ salt: _getL2Salt(OrbitSalts.L2_PROXY_ADMIN) }());
```

Comment:
Yep it fails in case contract already exists. Added custom error - https://github.com/OffchainLabs/token-bridge-contracts/pull/36/commits/9d8cc955507de5624fd1df1d0dde5e4f3043a8ef